lib/spec_test.go

```diff
@@ -10,10 +11,21 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
+func TestReadSpecMetadataMock(t *testing.T) {
+	spec, diags := ReadSpecMetadataMock("testdata/scenario.tfspec")
+	if diags.HasErrors() {
+		t.Fatal(diags.Err())
+	}
+
+	if spec == nil {
+		t.Fatal("spec is nil")
+	}
+}
+
 func TestParsing(t *testing.T) {
 	schemas := &terraform.Schemas{
-		Providers: map[string]*terraform.ProviderSchema{
-			"ressource": {
+		Providers: map[addrs.Provider]*terraform.ProviderSchema{
+			addrs.NewDefaultProvider("ressource"): {
```

Comment:
Seems like the pattern we want, however looking at the code, I think at the moment 3rd party providers are not supported. I don't have a use case at the moment personally but I think its a nice feature.